torchvision/datasets/vision.py

```diff
@@ -7,7 +9,17 @@
 from ..utils import _log_api_usage_once
 
 
-class VisionDataset(data.Dataset):
+class Hack(ABCMeta):
```

Comment:
We just need to subclass from `ABCMeta` here, since some of our datasets inherit from `ABC`:

https://github.com/pytorch/vision/blob/707457050620e1f70ab1b187dad81cc36a7f9180/torchvision/datasets/_optical_flow.py#L30

Without them, we could also inherit from `type` directly.

Reply:
Could we apply something like that to the `VisionDatasetDatapointWrapper` class instead of altering all the existing classes?

Reply:
I don't think so, no. `__instancecheck__` is only triggered for the second argument of the `instance` call. Meaning, we have no hooks for `isinstance(wrapped_mnist, datasets.MNIST)` unless we change something about `datasets.MNIST` or its base class `datasets.VisionDataset`.

The only other way to achieve that is to let our wrapper properly subclass. Let me give it a shot.